S/scopehal/bundled/patches/0001-replace-color-with-string.patch

```diff
@@ -0,0 +1,280 @@
+From a6816f4d663e759cb6e9f7c1af96e4171d6c0558 Mon Sep 17 00:00:00 2001
+From: Pepijn de Vos <pepijndevos@gmail.com>
+Date: Fri, 20 Aug 2021 11:04:37 +0200
+Subject: [PATCH] replace color with string
+
+---
+ .gitmodules                   |  6 +++---
+ scopehal/AlignedAllocator.h   | 11 ++++++++++-
+ scopehal/CMakeLists.txt       |  2 +-
+ scopehal/Filter.cpp           | 20 ++++++++++----------
+ scopehal/Filter.h             |  4 ++--
+ scopehal/MockOscilloscope.cpp |  1 +
+ scopehal/PacketDecoder.cpp    | 18 +++++++++---------
+ scopehal/PacketDecoder.h      |  6 +++---
+ scopehal/scopehal.cpp         |  7 ++++++-
+ scopehal/scopehal.h           | 13 +++++++++----
+ 10 files changed, 54 insertions(+), 34 deletions(-)
+
+diff --git a/.gitmodules b/.gitmodules
+index 86185b9..7acf360 100644
+--- a/.gitmodules
++++ b/.gitmodules
+@@ -1,12 +1,12 @@
+ [submodule "xptools"]
+ 	path = xptools
+-	url = ../xptools.git
++	url = https://github.com/azonenberg/xptools.git
+ [submodule "log"]
+ 	path = log
+-	url = ../logtools.git
++	url = https://github.com/azonenberg/logtools.git
+ [submodule "graphwidget"]
+ 	path = graphwidget
+-	url = ../graphwidget.git
++	url = https://github.com/azonenberg/graphwidget.git
+ [submodule "OpenCL-CLHPP"]
+ 	path = OpenCL-CLHPP
+ 	url = https://github.com/KhronosGroup/OpenCL-CLHPP.git
+diff --git a/scopehal/AlignedAllocator.h b/scopehal/AlignedAllocator.h
+index 10c2879..7d5eb85 100644
+--- a/scopehal/AlignedAllocator.h
++++ b/scopehal/AlignedAllocator.h
+@@ -38,6 +38,10 @@
+ 
+ #ifdef _WIN32
+ #include <windows.h>
++#elif __APPLE__
++#include <stdlib.h>
++#else
++#include <malloc.h>
```

Comment:
For what is worth, I believe `stdlib.h` is the right header file to include everywhere.  `malloc.h` was deprecate some ten years ago and it's Linux-specific.

Reply:
Well could be, but but according to https://linux.die.net/man/3/memalign it depends on the system, so whatever works.

Reply:
Oh, right, because also `memalign` is deprecated? :sweat_smile: